### PR TITLE
refactor(app): update JogControls component styling

### PR DESCRIPTION
**Pull request by sakibh**
# Overview

This PR updates the `JogControls` component styling for both the `StepSizeControl` and `DirectionControl` sub-components. closes [RAUT-84](https://opentrons.atlassian.net/browse/RAUT-84?atlOrigin=eyJpIjoiYmUzMmVkNjViOGE2NDE2ZDk4ZjgyYmRhYTYxMGNhMzgiLCJwIjoiaiJ9)

image

# Changelog

- Update styling in `StepSizeControl`, `DirectionControl`, and `ControlContainer`
- Add listener for `Shift` key press to switch axes to vertical/z-axis.
 
# Review requests

- Test the deck recalibration/pipette calibration flows on a real robot to ensure functionality is persistent.
- Press the shift key with the up and down arrow to test the ability to switch axes on the UI.
- Check the styling changes for further optimization

# Risk assessment
low, mostly stylistic

### CODE REVIEW

File: app/src/molecules/JogControls/ControlContainer.tsx [hunk 1]
```diff
@@ -1,29 +1,40 @@
...
-      <Text css={FONT_BODY_1_DARK} marginBottom={SPACING_2}>
-        {props.subtitle}
-      </Text>
-      {props.children}
+    <Flex flexDirection={DIRECTION_COLUMN}>
+      <StyledText
+        textTransform={TYPOGRAPHY.textTransformUppercase}
+        color={COLORS.darkGreyEnabled}
+        fontWeight={TYPOGRAPHY.fontWeightRegular}
+        fontSize={TYPOGRAPHY.fontSizeH6}
```
Comment (jerader) [inline app/src/molecules/JogControls/ControlContainer.tsx]:
```suggestion
        css={TYPOGRAPHY.h6Default}
```

File: app/src/molecules/JogControls/ControlContainer.tsx [hunk 2]
```diff
@@ -1,29 +1,40 @@
...
+        marginY={SPACING.spacing3}
+      >
+        {props.title}
+      </StyledText>
+      <Flex
+        backgroundColor={COLORS.fundamentalsBackground}
+        borderRadius={BORDERS.radiusSoftCorners}
+        padding={SPACING.spacing4}
+        width="365px"
+        height="156px"
```
Comment (jerader) [inline app/src/molecules/JogControls/ControlContainer.tsx]:
```suggestion
        width="22.8rem"
        height="9.75rem"
```

File: app/src/molecules/JogControls/DirectionControl.tsx [hunk 3]
```diff
@@ -53,13 +65,14 @@ const CONTROLS_CONTENTS_BY_PLANE: Record<Plane, ControlsContents> = {
...
+        margin: '1rem 0 0 1rem',
         iconName: 'ot-arrow-down',
         axis: 'z',
         sign: -1,
       },
     ],
-    title: 'Up & Down',
-    subtitle: 'Arrow keys + SHIFT',
+    title: 'Z-axis',
+    subtitle: 'Shift + Arrow keys',
```
Comment (jerader) [inline app/src/molecules/JogControls/DirectionControl.tsx:75]: I wonder if we should add these to constants?

File: app/src/molecules/JogControls/DirectionControl.tsx [hunk 4]
```diff
@@ -99,71 +115,230 @@ const CONTROLS_CONTENTS_BY_PLANE: Record<Plane, ControlsContents> = {
...
+        margin: '1rem 0 0 1rem',
         iconName: 'ot-arrow-down',
         axis: 'y',
         sign: -1,
       },
     ],
-    title: 'Across Deck',
-    subtitle: 'Arrow keys',
+    title: 'X- and Y-axis',
+    subtitle: 'Arrow Keys',
```
Comment (jerader) [inline app/src/molecules/JogControls/DirectionControl.tsx:125]: add these to constants too?

File: app/src/molecules/JogControls/DirectionControl.tsx [hunk 5]
```diff
@@ -99,71 +115,230 @@ const CONTROLS_CONTENTS_BY_PLANE: Record<Plane, ControlsContents> = {
...
-  const { title, subtitle, controls } = CONTROLS_CONTENTS_BY_PLANE[props.plane]
+  const { planes, jog, stepSize, defaultPlane } = props
+  const [currentPlane, setCurrentPlane] = React.useState<Plane>(
+    defaultPlane ?? planes[0]
+  )
+  const { t } = useTranslation(['robot_calibration'])
+
+  const DIRECTION_BUTTON_STYLING = css`
+    button {
+      background-color: white;
```
Comment (jerader) [inline app/src/molecules/JogControls/DirectionControl.tsx]:
```suggestion
      background-color: COLORS.white;
```

File: app/src/molecules/JogControls/DirectionControl.tsx [hunk 6]
```diff
@@ -99,71 +115,230 @@ const CONTROLS_CONTENTS_BY_PLANE: Record<Plane, ControlsContents> = {
...
-                  width="100%"
+          {planes.map((plane: Plane) => {
+            const { title, subtitle } = CONTROLS_CONTENTS_BY_PLANE[plane]
+            return (
+              <Flex key={plane}>
+                <PrimaryButton
+                  title={plane}
+                  css={currentPlane === plane ? ACTIVE_STYLE : DEFAULT_STYLE}
+                  minWidth="157px"
+                  height="58px"
```
Comment (jerader) [inline app/src/molecules/JogControls/DirectionControl.tsx]:
```suggestion
                  minWidth="9.81rem"
                  height="3.62rem"
```

File: app/src/molecules/JogControls/DirectionControl.tsx [hunk 7]
```diff
@@ -99,71 +115,230 @@ const CONTROLS_CONTENTS_BY_PLANE: Record<Plane, ControlsContents> = {
...
+  stepSize: number
+}
+
+export const ArrowKeys = (props: ArrowKeysProps): JSX.Element => {
+  const { plane, jog, stepSize } = props
+  const controls = CONTROLS_CONTENTS_BY_PLANE[plane].controls
+
+  const ARROW_BUTTON_STYLING = css`
+    button {
+      background-color: white;
```
Comment (jerader) [inline app/src/molecules/JogControls/DirectionControl.tsx]:
```suggestion
      background-color: COLORS.white;
```

File: app/src/molecules/JogControls/DirectionControl.tsx [hunk 8]
```diff
@@ -99,71 +115,230 @@ const CONTROLS_CONTENTS_BY_PLANE: Record<Plane, ControlsContents> = {
...
+        background-color: ${COLORS.white};
+        color: ${COLORS.darkGreyHover};
+        box-shadow: 0 0 0;
+        border: 1px ${COLORS.lightGreyHover} solid;
+      }
+
+      &:active {
+        background-color: ${COLORS.white};
+        color: ${COLORS.darkGreyPressed};
+        border: 1px #16212d40 solid;
```
Comment (jerader) [inline app/src/molecules/JogControls/DirectionControl.tsx]:
```suggestion
        border: 1px ${COLORS.lightGreyHover} solid;
```

File: app/src/molecules/JogControls/StepSizeControl.tsx [hunk 9]
```diff
@@ -34,12 +42,59 @@ export function StepSizeControl(props: StepSizeControlProps): JSX.Element {
...
+  }
+
+  const JUMP_SIZE_ICON_STYLE = css`
+    flex-shrink: 0;
+  `
+
+  const STEP_SIZE_LABELS: { [stepSize: number]: string } = {
+    0.1: 'Tiny',
+    1: 'Small',
+    10: 'Large',
```
Comment (jerader) [inline app/src/molecules/JogControls/StepSizeControl.tsx]: these should be added to i18n

File: app/src/molecules/JogControls/StepSizeControl.tsx [hunk 10]
```diff
@@ -34,12 +42,59 @@ export function StepSizeControl(props: StepSizeControlProps): JSX.Element {
...
   }
+
+  const STEP_SIZE_BUTTON_STYLING = css`
+    button {
+      background-color: white;
```
Comment (jerader) [inline app/src/molecules/JogControls/StepSizeControl.tsx]:
```suggestion
      background-color: COLORS.white;
```

File: app/src/molecules/JogControls/StepSizeControl.tsx [hunk 11]
```diff
@@ -49,16 +104,54 @@ export function StepSizeControl(props: StepSizeControlProps): JSX.Element {
...
-          }))}
-          onChange={handleStepSelect}
-          className={lpcRadiobuttonColor}
-        />
+        <Flex flexDirection={DIRECTION_COLUMN}>
+          <Flex flexDirection={DIRECTION_ROW}>
+            <Icon name="jump-size" width="1.2rem" css={JUMP_SIZE_ICON_STYLE} />
+            <StyledText
+              textTransform={TEXT_TRANSFORM_CAPITALIZE}
+              color={COLORS.black}
```
Comment (jerader) [inline app/src/molecules/JogControls/StepSizeControl.tsx]: `StyledText` adds `black` as the default text color so you dont need to redefine it

```suggestion
```

File: app/src/organisms/LabwarePositionCheck/LabwarePositionCheckStepDetail.tsx [hunk 12]
```diff
@@ -223,7 +223,6 @@ export const LabwarePositionCheckStepDetail = (
             planes={[HORIZONTAL_PLANE, VERTICAL_PLANE]}
             width="100%"
             directionControlButtonColor={COLORS.blueEnabled}
-            isLPC={true}
```
Comment (jerader) [inline app/src/organisms/LabwarePositionCheck/LabwarePositionCheckStepDetail.tsx:226]: 🎉 so cool that we can remove this now

File: app/src/molecules/JogControls/StepSizeControl.tsx [hunk 13]
```diff
@@ -34,12 +42,59 @@ export function StepSizeControl(props: StepSizeControlProps): JSX.Element {
...
+    flex-shrink: 0;
+  `
+
+  const STEP_SIZE_LABELS: { [stepSize: number]: string } = {
+    0.1: 'Tiny',
+    1: 'Small',
+    10: 'Large',
   }
+
+  const STEP_SIZE_BUTTON_STYLING = css`
```
Comment (jerader) [inline app/src/molecules/JogControls/StepSizeControl.tsx:62]: Are these Button stylings all the same for this component and `DirectionControl`? If so, I suggest exporting the constant in `DirectionControl` and then importing it here so they can share the same constant.

**Review by jerader — CHANGES_REQUESTED** [review]
Its looking good so far! I left a bunch of styling suggestions. The only thing missing is the responsiveness and I think it would be nice to add `DirectionControl` and `StepSizeControl` to Storybook

**Review by jerader — APPROVED** [review]
this is looking great! Good job! 🚀 The story looks good and responsive. The only thing is do we want to center the contents to the grey container? 

Screen Shot 2022-08-22 at 3 16 41 PM

**Comment by sakibh** [comment]
> 

I think this is because storybook has no upper bound for the width, whereas the app does. It might be better to make the buttons itself resize with the container. I can address this in a follow up.

**Pull request merged**